python/lsst/daf/persistence/mapper.py

```diff
@@ -96,6 +97,7 @@ def __new__(cls, *args, **kwargs):
         return self
 
     def __init__(self, **kwargs):
+        deprecateGen2("Mapper")
```

Comment:
Have we checked that all `Mapper` subclasses call `super().__init__`?

Reply:
No. To a certain extent instantiating a `Butler` will get most places.

Reply:
I can add an `__init_subclass__` to `Mapper` to deal with that I think.

Reply:
Of course `__init_subclass__` doesn't help us because that triggers at import and not object construction. Maybe I have to insert a `__new__`?

Reply:
Rewrote to use a class decorator.